once_test.py

```diff
@@ -478,30 +607,45 @@ def receiving_iterator():
         self.assertEqual(list(receiving_iterator()), [0, 1, 2, 5])
 
     def test_receiving_iterator_parallel_execution(self):
+        # Pause so we actually are able to test parallel execution, by ensuring that each exec
+        # does not complete before the next one is scheduled.
+        paused = True
+
         @once.once
         def receiving_iterator():
+            nonlocal paused
             next = yield 0
             while next is not None:
+                while paused:
+                    pass
                 next = yield next
 
         def call_iterator(_):
             gen = receiving_iterator()
             result = []
             result.append(gen.send(None))
-            for i in range(1, 32):
+            for i in range(1, _N_WORKERS * 4):
                 result.append(gen.send(i))
             return result
 
-        with concurrent.futures.ThreadPoolExecutor(max_workers=16) as executor:
-            results = list(executor.map(call_iterator, range(32)))
+        with concurrent.futures.ThreadPoolExecutor(max_workers=_N_WORKERS) as executor:
+            results = executor.map(call_iterator, range(_N_WORKERS * 2))
+            paused = False  # starter pistol, the race is off!
```

Comment:
I think you actually want a barrier inside call_iterator instead

Reply:
Only one iterator will actually be instantiated by once though :/